### PR TITLE
Fix `runtime-import-check` errors

### PR DESCRIPTION
**Pull request by alvsan09**
#### What it does

Fixes `runtime-import-check` errors related to `plugin-api-rpc`
as well as from the re-use of `DebugPluginConfiguration`.

This is done by moving relevant code under their proper runtime (browser vs common vs node).
For more details see: #10418

#### How to test

There should not be a change in functionality:

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 - [Previous Changelogs](https://github.com/eclipse-theia/theia/tree/master/doc/changelogs/)
 
+<a name="breaking_changes_1.26.0">[Breaking Changes:](#breaking_changes_1.26.0)</a>
+- Moved interface `DebugPluginConfiguration`
+from  `@theia/debug/lib/browser/debug-contribution`  
+to    `@theia/debug/lib/common/debug-configuration`
```
Comment (vince-fugnitto) [inline CHANGELOG.md:10]: Do you mind updating to the following?

```suggestion
## v1.27.0 - Unreleased

<a name="breaking_changes_1.27.0">[Breaking Changes:](#breaking_changes_1.27.0)</a>

- [plugin] moved interface `DebugPluginConfiguration` from `@theia/debug/lib/browser/debug-contribution` to `@theia/debug/lib/common/debug-configuration` [#11224](https://github.com/eclipse-theia/theia/pull/11224)
```

Comment (alvsan09) [reply CHANGELOG.md]: Done, thanks Vince !!

File: packages/debug/src/common/debug-configuration.ts [hunk 2]
```diff
@@ -84,6 +84,12 @@ export namespace DebugConfiguration {
     }
 }
 
+export interface DebugPluginConfiguration {
+    debugMode?: string;
+    pluginLocation?: string;
+    debugPort?: string;
+}
```
Comment (vince-fugnitto) [inline packages/debug/src/common/debug-configuration.ts:91]: Any idea why this interface lives in `debug`? 
It seems pretty specific to `plugin-dev` (for hosted-plugins) and not ultimately used by `debug`.

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me :+1: 

I confirmed that:
- [x] renaming `DebugPluginConfiguration` to `PluginDebugConfiguration` makes sense
- [x] moving `PluginDebugConfiguration` to `plugin-dev` over `debug` is better since it is plugin-specific
- [x] hosted-plugin debugging works (aside from known issue https://github.com/eclipse-theia/theia/issues/7652)
- [x] custom-editors work well (using cat example)

**Pull request merged**